### ISSUE TITLE
tarball has tests in tests/tests

### ISSUE THREAD

**Issue by zeha** [shadow-maint/shadow#1027]
The 4.15.2 and the 4.16.0 tarballs have a surprising layout for the test files:

```
shadow-4.16.0/tests
shadow-4.16.0/tests/unit
shadow-4.16.0/tests/unit/test_xasprintf.c
shadow-4.16.0/tests/unit/test_zustr2stp.c
shadow-4.16.0/tests/unit/test_logind.c
shadow-4.16.0/tests/unit/test_strncpy.c
shadow-4.16.0/tests/unit/Makefile.am
shadow-4.16.0/tests/unit/test_strtcpy.c
shadow-4.16.0/tests/unit/test_adds.c
shadow-4.16.0/tests/unit/test_atoi_strtoi.c
shadow-4.16.0/tests/unit/test_chkname.c
shadow-4.16.0/tests/unit/Makefile.in
shadow-4.16.0/tests/unit/test_sprintf.c
shadow-4.16.0/tests/tests
shadow-4.16.0/tests/tests/cktools
...
shadow-4.16.0/tests/tests/unit
shadow-4.16.0/tests/tests/unit/test_xasprintf.c
shadow-4.16.0/tests/tests/unit/test_zustr2stp.c
shadow-4.16.0/tests/tests/unit/Makefile
shadow-4.16.0/tests/tests/unit/test_logind.c
shadow-4.16.0/tests/tests/unit/test_strncpy.c
shadow-4.16.0/tests/tests/unit/Makefile.am
shadow-4.16.0/tests/tests/unit/test_strtcpy.c
shadow-4.16.0/tests/tests/unit/.gitignore
shadow-4.16.0/tests/tests/unit/test_adds.c
shadow-4.16.0/tests/tests/unit/test_atoi_strtoi.c
shadow-4.16.0/tests/tests/unit/Makefile.in
shadow-4.16.0/tests/tests/unit/test_sprintf.c
...
```

So at least tests/unit is shipped twice, and IMO it shouldn't be in a different directory than in git.

Could this just become like in git, please?

**Comment by alejandro-colomar** [comment]
@hallyn

**Comment by hallyn** [comment]
Ah, this is probably because tests/unit is in SUBDIRS in Makefile.am, since before tests/ was in EXTRA_DIST.  We should probably just drop that from SUBDIRS.

**Comment by alejandro-colomar** [comment]
Hi Serge,

On Wed, Jun 26, 2024 at 08:18:57PM GMT, Serge Hallyn wrote:
> Ah, this is probably because tests/unit is in SUBDIRS in Makefile.am, since before tests/ was in EXTRA_DIST.  We should probably just drop that from SUBDIRS.

I've prepared a patch with what I think I understood from here:
<https://github.com/shadow-maint/shadow/pull/1030>

However, that doesn't build correctly:
<https://github.com/shadow-maint/shadow/actions/runs/9693463976/job/26748961956?pr=1030>

	config.status: error: cannot find input file: `tests/unit/Makefile.in'

I'll wait for that one to be fixed, and then will release 4.15.3.

Please suggest any alternative ideas, and I'll write them.

Cheers,
Alex

> -- 
> Reply to this email directly or view it on GitHub:
> https://github.com/shadow-maint/shadow/issues/1027#issuecomment-2193217830
> You are receiving this because you commented.
> 
> Message ID: ***@***.***>

-- 
<https://www.alejandro-colomar.es/>
